### ISSUE TITLE
Add extension point for a custom filtering stage

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#3518]
There are use cases like this:
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/dita-ot-users/u6sGODXRCrw/omNkIWHmAwAJ

when filtering based on the DITAVAL file is not enough.
From what I remember, at some point @jelovirt had this idea of adding the possibility for a custom script (javascript for example) to be used in addition to the DITAVAL file in the filtering stage in order to make more complex filtering decisions.

**Comment by jelovirt** [comment]
Yes, I was toying with the idea to allow users to do `--filter=custom.js` and that would be something like

```js
function filter(attrs) {
  // custom filtering based on the attrs object.
}
```

**Comment by raducoravu** [comment]
Maybe receiving more details on the callback, like the location of the parsed topic would have helped to resolve relative references in attribute values for example.
But yes, such a  custom filter stage would be useful also for people who filter based on DITA profiling attributes but they want some more complex logic, like excluding an element only when it has two profiling attributes, each with its own specific value.

**Comment by raducoravu** [comment]
Leaving here a discussion from the write the docs dita slack channel: 

But can we filter based on three different values of the same attribute? That's what we're trying to do, I think.

Is it possible to
1. include all untagged (common) content
2. include all stuff that's tagged product=A AND product=C
3. include all the stuff that's tagged product=B AND product=C, but 
4. exclude all the stuff that's only tagged product=A or product=B? (or, I suppose, in the future, product=A, B, D, and E but NOT C)
